src/nvim/option.c

```diff
@@ -1804,19 +1804,20 @@ void set_init_1(void)
 # endif
     int len;
     garray_T ga;
-    bool mustfree;
 
     ga_init(&ga, 1, 100);
     for (size_t n = 0; n < ARRAY_SIZE(names); ++n) {
-      mustfree = FALSE;
+      bool mustfree = true;
```

Comment:
Here in this function we have an ifdef that leverages using the bracketless if that affects the way the function behaves.  Within the ifdef it sets the pointer to a string constant, which creates a problem when trying to free later.  I setup a mustfree variable in the local scope to handle this, but I *could* also do something like:
```
#ifndef UNIX
        xfree(p);
#endif
```

If there's a preference for that let me know and I'll change it.